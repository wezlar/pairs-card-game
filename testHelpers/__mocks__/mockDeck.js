const mockDeck = [
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
  {
    cardIndex: 0,
    hasMatched: false,
    image: 'images/svg-cards/2_of_hearts.svg',
    isFlipped: false,
    name: '2 of hearts',
  },
  {
    cardIndex: 1,
    hasMatched: false,
    image: 'images/svg-cards/3_of_hearts.svg',
    isFlipped: false,
    name: '3 of hearts',
  },
  {
    cardIndex: 2,
    hasMatched: false,
    image: 'images/svg-cards/4_of_hearts.svg',
    isFlipped: false,
    name: '4 of hearts',
  },
];

export default mockDeck;