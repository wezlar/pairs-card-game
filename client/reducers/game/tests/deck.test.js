import actionCreator from '../../../../testHelpers/actionCreator';

import ACTION_TYPES from '../../../actions/constants';
import reducer, { getInitialState } from '../deck';

import mockDeck from '../../../../testHelpers/__mocks__/mockDeck';

describe(`deck.js reducer`, () => {
  const initialState = getInitialState();

  describe('initial state', () => {
    test('it matches the snapshot - default', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe(`the reducer`, () => {
    test('it returns initial state by default', () => {
      expect(reducer(undefined, {})).toEqual(initialState)
    });
    
    test(`it doesn't change state unexpectedly`, () => {
      const fakeAction = actionCreator('NOT_A_REAL_ACTION_TYPE')('test-payload');
      expect(reducer(undefined, fakeAction)).toEqual(initialState)
    });

    test(`status is correct for ${ACTION_TYPES.ADD_NEW_DECK}`, () => {
      const payload = { deck: mockDeck };
      const action = actionCreator(ACTION_TYPES.ADD_NEW_DECK)(payload)
      expect(reducer(undefined, action)).toEqual(payload.deck);
    });

    test(`status is correct for ${ACTION_TYPES.CARD_FLIP}`, () => {
      const mockInitialState = mockDeck;
      const payload = { index: 5 };
      const action = actionCreator(ACTION_TYPES.CARD_FLIP)(payload)
      expect(reducer(mockInitialState, action)[0].isFlipped).toEqual(false);
      expect(reducer(mockInitialState, action)[5].isFlipped).toEqual(true);
    });

    test(`status is correct for ${ACTION_TYPES.CARD_MATCHED}`, () => {
      const mockInitialState = mockDeck;
      const payload = { index: 3 };
      const action = actionCreator(ACTION_TYPES.CARD_MATCHED)(payload)
      expect(reducer(mockInitialState, action)[1].hasMatched).toEqual(false);
      expect(reducer(mockInitialState, action)[3].hasMatched).toEqual(true);
    });
  });
});