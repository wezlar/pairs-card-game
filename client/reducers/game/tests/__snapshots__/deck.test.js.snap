// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deck.js reducer initial state it matches the snapshot - default 1`] = `Array []`;
